gsa/src/web/pages/usersettings/usersettingspage.js

```diff
@@ -368,6 +368,48 @@ class UserSettings extends React.Component {
       autoCacheRebuild = {},
     } = this.props;
 
+    // TODO replace the following with nullish coalescent operator once available
+    agentsFilter = isNull(agentsFilter) ? {} : agentsFilter;
```

Comment:
You should drop the defaults above and use the following code instead. We should only set the expected value once.
```suggestion
    agentsFilter = hasValue(agentsFilter) ? {} : agentsFilter;
```

Reply:
This is not equivalent. hasValue() explicitly checks for the variable to be NOT null

Reply:
Sorry should be
```javascript
 agentsFilter = hasValue(agentsFilter) ? agentsFilter : {};
```
of course.